Sources/Parsing/Parsers/End.swift

```diff
@@ -8,6 +8,14 @@ public struct End<Input>: Parser where Input: Collection {
   @inlinable
   public init() {}
 
+  @_disfavoredOverload
+  @inlinable
+  public init() where Input == Substring {}
+
+  @_disfavoredOverload
+  @inlinable
+  public init() where Input == Substring.UTF8View {}
```

Comment:
Parsers beyond `Bool.parser()` and `Int.parser()` can benefit from this default inference, as well.